gateway-service/src/test/java/com/ca/mfaas/gateway/filters/post/PageRedirectionFilterTest.java

```diff
@@ -0,0 +1,333 @@
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.springframework.cloud.netflix.zuul.filters.support.FilterConstants.SERVICE_ID_KEY;
+
+public class PageRedirectionFilterTest {
+
+    private static final String SERVICE_ID = "discovered-service";
+    private static final String TARGET_SERVER_HOST = "hostA.test.com";
+    private static final int TARGET_SERVER_PORT = 8888;
+    private static final String OTHER_SERVICE_ID = "other-service";
+    private static final String OTHER_SERVICE_SERVER_HOST = "hostB.test.com";
+    private static final int OTHER_SERVICE_SERVER_PORT = 9999;
+    private static final String NOT_IN_DS_SERVER_HOST = "hostC.test.com";
+    private static final int NOT_IN_DS_SERVER_PORT = 7777;
+
+    private PageRedirectionFilter filter = null;
+    private DiscoveryClient discoveryClient = null;
+    private MockHttpServletResponse response = null;
+
+    @Before
+    public void setUp() {
+        discoveryClient = mock(DiscoveryClient.class);
+
+        RequestContext ctx = RequestContext.getCurrentContext();
+        ctx.clear();
+        ctx.set(SERVICE_ID_KEY, SERVICE_ID);
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        response = new MockHttpServletResponse();
+        ctx.setRequest(request);
+        ctx.setResponse(response);
+
+        this.filter = new PageRedirectionFilter(this.discoveryClient, ctx.getRequest().getLocalName(), String.valueOf(ctx.getRequest().getLocalPort()), ctx.getRequest().getScheme());
+    }
+
+    /**
+     * <ul>Test the Location url which
+     * <li>contains the same host and port with current service</li>
+     * <li>can be matched to gateway url</li>
+     * </ul>
+     *
+     * @throws Exception
+     */
+    @Test
+    public void sameServerAndUrlMatched() throws Exception {
+        RoutedService currentService = new RoutedService("ui", "ui", "/");
+        RoutedServices routedServices = new RoutedServices();
+        routedServices.addRoutedService(currentService);
+        this.filter.addRoutedServices(SERVICE_ID, routedServices);
+
+        when(discoveryClient.getInstances(SERVICE_ID)).thenReturn(Arrays.asList(
+            new DefaultServiceInstance(SERVICE_ID, TARGET_SERVER_HOST, TARGET_SERVER_PORT, true)
+        ));
+
+        response.setStatus(302);
+        String relativePath = "/some/path/login.html";
+        String location = mockLocationSameServer(relativePath);
+        final RequestContext ctx = RequestContext.getCurrentContext();
+        ctx.addZuulResponseHeader(LOCATION, location);
+        this.filter.run();
+
+        Optional<Pair<String, String>> locationHeader = ctx.getZuulResponseHeaders()
+            .stream()
+            .filter(stringPair -> LOCATION.equals(stringPair.first()))
+            .findFirst();
+
+        verifyLocationUpdatedSameServer(locationHeader.get().second(), location,
```

Comment:
```suggestion
        verifyLocationUpdatedSameServer(locationHeader.map(Pair::second).orElse(null), location,
```
Call get() on Optional without checking on is present not recommended. And this in all methods below in this class.

Reply:
Thanks for comments, fixed in https://github.com/zowe/api-layer/pull/276/commits/8df2424832616a6c2a60f7a3d96aed5e0c80f9a3